code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.zuul/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/zuul/code_generation/ZuulCodeGenerator.xtend

```diff
@@ -0,0 +1,131 @@
+     * The corresponding functions are called depending on whether the operation node is an instance
+     * of an intermediate container or infrastructure node.
+     */
+    override execute(String[] phaseArguments, String[] moduleArguments) {
+        // Receive the intermediate operation model
+        model = intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create infrastructure node-specific Zuul components
+        model.infrastructureNodes.forEach[
+            val usesZuulTechnology =
+                it.qualifiedInfrastructureTechnologyName.toLowerCase == ZUUL_TECHNOLOGY
+
+            if (usesZuulTechnology) {
+                createEurekaKubernetesDeployment(it)
+                createDockerComposeDeployment(it)
+                createZuulComponents(it)
+            }
+        ]
+
+        return withCharset(content, "UTF-8");
+    }
+
+    /**
+     * Receive the language description for the intermediate operation model package.
+     */
+    @NotNull
+    override getLanguageDescription() {
+        new LanguageDescription(IntermediatePackage.eINSTANCE)
+    }
+
+    /**
+     * Create an infrastructure node-specific Kubernetes deployment file for Zuul.
+     */
+    private def createEurekaKubernetesDeployment(IntermediateInfrastructureNode node) {
+        val kubernetes = ZuulTemplate::getKubernetesDeploymentForZuul(node)
+        val path = '''«targetFolder»«File.separator»«node.name»«File.separator»''' +
+            '''«node.name.toLowerCase»-deployment.yaml'''
+
+        content.put(path, kubernetes)
+    }
+
+    /**
+     * Create an  infrastructure node-specific Docker-Compose deployment file for Zuul.
+     */
+    private def createDockerComposeDeployment(IntermediateInfrastructureNode node) {
+        /*
+         * Set targetFolder for Docker-Compose file and open an existing or create a new
+         * Docker-Compose file
+         */
+        OpenedDockerComposeFile.instance.init(targetFolder)
+        OpenedDockerComposeFile.instance.openExistingDockerComposeFile
+
+        // Build Docker-Compose part based on a operation aspect or node configuration
+        val dockerComposeAspect = node.aspects.findFirst[aspect | aspect.name == "ComposePart"]
+        if (dockerComposeAspect === null)
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                ZuulTemplate::getDockerComposeForZuul(node))
+        else
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                dockerComposeAspect.propertyValues?.get(0).value)
+
+        // Write Docker-Compose file to the file path
+        val filePath = '''«targetFolder»«File.separator»docker-compose.yml'''
+        content.put(filePath, OpenedDockerComposeFile.instance.toString)
+    }
+
+    /**
+     * Generate a Zuul infrastructure node based on the Spring Cloud framework implemented in
+     * Kotlin.
```

Comment:
What was implemented in Kotlin? Spring Cloud? Is this information really necessary? If so, please formulate it more precisely.